sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MapStateImplWithTTL.scala

```diff
@@ -169,13 +174,22 @@ class MapStateImplWithTTL[K, V](
    * @param groupingKey grouping key for which cleanup should be performed.
    * @param userKey     user key for which cleanup should be performed.
    */
-  override def clearIfExpired(groupingKey: Array[Byte], userKey: Array[Byte]): Long = {
-    val encodedCompositeKey = stateTypesEncoder.encodeCompositeKey(groupingKey, userKey)
-    val retRow = store.get(encodedCompositeKey, stateName)
+  override def clearIfExpired(
+      groupingKeyRow: UnsafeRow,
+      userKeyRow: UnsafeRow): Long = {
+    val compositeKeyRow =
```

Comment:
we need to do this each time ?

Reply:
Put the projection into a new function in `StateTypesEncoderUtils` to avoid initializing projection multiple times